src/lib.rs

```diff
@@ -770,41 +804,46 @@ mod test {
   }
 
   #[test]
-  fn check_large_file_size_over_11_chunks() {
-    // has been tested for 50 chunks
+  fn check_10_and_a_bit_max_chunks() {
     let mut my_storage = MyStorage::new();
     let mut data_map = datamap::DataMap::None;
-    let number_of_chunks = 11 as u32;
-    let string_len = (MAX_CHUNK_SIZE as u64 * number_of_chunks as u64) + 1024;
+    let string_len = (MAX_CHUNK_SIZE as u64 * 10) + 1024;
     let the_string = random_string(string_len);
     {
       let mut se = SelfEncryptor::new(&mut my_storage as &mut Storage, datamap::DataMap::None);
       se.write(&the_string, 0);
-      assert_eq!(se.get_num_chunks(), number_of_chunks + 1);
-      for i in 0u32..number_of_chunks {
-        // preceding and next index, wrapped around
-        let h = (i + number_of_chunks)%(number_of_chunks + 1) as u32;
-        let j = (i + 1)%(number_of_chunks + 1) as u32;
-        assert_eq!(se.get_chunk_size(i), MAX_CHUNK_SIZE);
-        assert_eq!(se.get_next_chunk_number(i), j);
-        assert_eq!(se.get_previous_chunk_number(i), h);
-        assert_eq!(se.get_start_end_positions(i).0, i as u64 * MAX_CHUNK_SIZE as u64);
-        assert_eq!(se.get_start_end_positions(i).1, j as u64 * MAX_CHUNK_SIZE as u64);
-      }
-      assert_eq!(se.get_chunk_size(number_of_chunks), MIN_CHUNK_SIZE);
-      assert_eq!(se.get_next_chunk_number(number_of_chunks), 0);
-      assert_eq!(se.get_previous_chunk_number(number_of_chunks), number_of_chunks - 1);
-      assert_eq!(se.get_start_end_positions(number_of_chunks).0, 
-                        number_of_chunks as u64 * MAX_CHUNK_SIZE as u64);
-      assert_eq!(se.get_start_end_positions(number_of_chunks).1, 
-                        ((number_of_chunks * MAX_CHUNK_SIZE) as u64 + 1024));
+      assert_eq!(se.get_num_chunks(), 11);
+      assert_eq!(se.get_chunk_size(0), MAX_CHUNK_SIZE);
```

Comment:
Here you explicitly replace a generalised loop with an incomplete, unordered set of assertions.  Not only missing out some iterations, but equally the final chunk which is a special case.